File: producer/models/notification.py
from pydantic import BaseModel


class Notification(BaseModel):
    providers: set
    body: str
    receivers: set


def notification_dict(notification) -> dict:
    return {
        "id": str(notification["_id"]),
        "providers": notification["providers"],
        "receivers": notification["receivers"],
        "body": notification["body"],
        "created_at": notification["created_at"],
    }

File: notification/models/notification.ts
import mongoose, { Schema } from 'mongoose';
import {NotificationDoc} from "./notificationDoc";

const NotificationSchema: Schema = new Schema({
    providers: { type: Array, required: true },
    body: { type: String, required: true },
    receivers: { type: Array, required: true },
    created_at: { type: String, required: true }
});
export const Notification = mongoose.model<NotificationDoc>('notification', NotificationSchema);

